pkg/k8s/openebs.go

```diff
@@ -119,18 +119,26 @@ func CreateCStorPoolClusters(captenConfig config.CaptenConfig) error {
 	return err
 }
 
-func CreateCStorPoolClusterWithRetries(captenConfig config.CaptenConfig) error {
-	retries := 3
-	for i := 0; i < retries; i++ {
-		err := CreateCStorPoolClusters(captenConfig)
+func RetryOperation(maxRetries int, retryInterval time.Duration, operation func() error) error {
+	var err error
+	for i := 0; i < maxRetries; i++ {
+		err = operation()
 		if err == nil {
-			clog.Logger.Info("Configured storage pool")
 			return nil
 		}
-		if i < retries-1 {
-
-			time.Sleep(5 * time.Second)
+		if i < maxRetries-1 {
+			time.Sleep(retryInterval)
 		}
 	}
-	return errors.New("Failed to configure storage pool after multiple retries")
+	return err
+}
+
```

Comment:
use this

func retry(retries int, interval time.Duration, f func() error) (err error) {
        for i := 0; i <= retries; i++ {
                if err = f(); err == nil {
					return nil
				}
                time.Sleep(interval) 
                return
        }
        return
}